DS3: Update setup documentation (#4437)
ArchipelagoMW/Archipelago: worlds/dark_souls_3/docs/setup_en.md

[.NET Runtime]: https://dotnet.microsoft.com/en-us/download/dotnet/8.0
[WINE]: https://www.winehq.org/

## Troubleshooting

### Enemy randomizer issues

> The DS3 Archipelago randomizer uses [thefifthmatt's DS3 enemy randomizer]

Reviewer: ```suggestion
The DS3 Archipelago randomizer uses [thefifthmatt's DS3 enemy randomizer],
```

essentially unchanged. Unfortunately, this randomizer has a few known issues,
including enemy AI not working, enemies spawning in places they can't be killed,

> and in a few rare cases enemies spawning in ways that crash the game when they

Reviewer: ```suggestion
and, in a few rare cases, enemies spawning in ways that crash the game when they
```

load. These bugs should be [reported upstream], but unfortunately the
Archipelago devs can't help much with them.

[thefifthmatt's DS3 enemy randomizer]: https://www.nexusmods.com/darksouls3/mods/484
[reported upstream]: https://github.com/thefifthmatt/SoulsRandomizers/issues

Because in rare cases the enemy randomizer can cause seeds to be impossible to
complete, we recommend disabling it for large async multiworlds for safety
purposes.

### `launchmod_darksouls3.bat` isn't working

> Sometimes `launchmod_darksouls3.bat` will briefly flash a terminal onscreen and

Reviewer: Could do "an onscreen terminal" also
```suggestion
Sometimes `launchmod_darksouls3.bat` will briefly flash a terminal on your screen and
```

then terminate without actually starting the game. This is usually caused by
some issue communicating with Steam either to find `DarkSoulsIII.exe` or to
launch it properly. If this is happening to you, make sure:

* You have DS3 1.15.2 installed. This is the latest patch as of January 2025.
  (Note that older versions of Archipelago required an older patch, but that
  _will not work_ with the current version.)

* You own the DS3 DLC if your randomizer config has DLC enabled. (It's possible

>   but unconfirmed that you need the DLC even when it's disabled in your config).

Reviewer: Is this really unconfirmed? I'm kinda surprised if _nobody_ has done this recently?

Author: I don't know of any situations where we had someone positively say "I had DLC disabled in my config and buying it fixed the crash", and I _really_ don't want to make it seem like a more likely fix than we're confident it is because it involves players spending real money.

Reviewer: I think the parenthetical reads as "We're not sure if you need the DLC to play the game even if you don't have them enabled" which sounds, idk, just like a bad thing to be unsure about?

Author: But we are actually unsure about it, and it's hard to test because you can't really _un_-buy the DLC.

Reviewer: Damn, that just kinda sucks, but if that's the way that it is, not really anything to do about it

> * You own the DS3 DLC if your randomizer config has DLC enabled. (It's possible
>   but unconfirmed that you need the DLC even when it's disabled in your config).

Reviewer: ```suggestion
* You own the DS3 DLC if your randomizer config has DLC enabled. (It's possible,
  but unconfirmed, that you need the DLC even when it's disabled in your config).
```

* Steam is not running in administrator mode. To fix this, right-click
  `steam.exe` (by default this is in `C:\Program Files\Steam`), select
  "Properties", open the "Compatiblity" tab, and uncheck "Run this program as an

>   administrator").

Reviewer: ```suggestion
  administrator".
```

* There is no `dinput8.dll` file in your DS3 game directory. This is the old way
  of installing mods, and it can interfere with the new ModEngine2 workflow.

If you've checked all of these, you can also try:

* Running `launchmod_darksouls3.bat` as an administrator.

* Reinstalling DS3 or even reinstalling Steam itself.

> * Making sure DS3 is installed on the same drive as steam and as the randomizer.

Reviewer: ```suggestion
* Making sure DS3 is installed on the same drive as Steam and as the randomizer.
```

  (A number of users are able to run these on different drives, but this has
  helped some users.)

If none of this works, unfortunately there's not much we can do. We use
ModEngine2 to launch DS3 with the Archipelago mod enabled, but unfortunately

> it's no longer maintained and its successor ModEngine3 isn't usable yet.

Reviewer: ```suggestion
it's no longer maintained and its successor, ModEngine3, isn't usable yet.
```
